src/software/simulation/er_force_simulator.cpp

```diff
@@ -67,77 +67,112 @@ ErForceSimulator::ErForceSimulator(
 
     // TODO (#2283): remove this initialization when addYellowRobots and addBlueRobots are
     // implemented
-    auto* team_blue   = simulator_setup_command->mutable_set_team_blue();
+
+    // TODO (#2283): remove this initialization when addYellowRobots and addBlueRobots are
+    // implemented
+
+    this->resetCurrentFirmwareTime();
+}
+
+void ErForceSimulator::setBallState(const BallState& ball_state)
+{
+    er_force_sim->safelyTeleportBall(static_cast<float>(ball_state.position().x()),
+                                     static_cast<float>(ball_state.position().y()));
+}
+
+void ErForceSimulator::addYellowRobots(const std::vector<RobotStateWithId>& robots)
+{
+    // TODO (#2283): add robots
+
+    auto simulator_setup_command = std::make_shared<amun::Command>();
+    simulator_setup_command->mutable_simulator()->set_enable(true);
+
+    robot::Specs ERForce;
+    robotSetDefault(&ERForce);
+
     auto* team_yellow = simulator_setup_command->mutable_set_team_yellow();
-    for (auto* team : {team_blue, team_yellow})
+    std::vector<std::tuple<int, int>>
+        yellow_robot_pos;  //=new std::vector<std::tuple<int, int>>;
+
+
+    for (unsigned int i = 0; i < robots.size(); i++)
     {
-        for (int i = 0; i < 11; ++i)
-        {
-            auto* robot = team->add_robot();
-            robot->CopyFrom(ERForce);
-            robot->set_id(i);
-        }
+        auto* robot = team_yellow->add_robot();
+        robot->CopyFrom(ERForce);
+        robot->set_id(i);
+        yellow_robot_pos.push_back(std::make_tuple(robots[i].robot_state.position().y(),
+                                                   robots[i].robot_state.position().x()));
```

Comment:
see if you could use the `TeleportRobot` message defined in `extlibs/er_force_sim/src/protobuf/ssl_simulation_control.proto` in a similar way to https://github.com/UBC-Thunderbots/Software/pull/2356/files#diff-abbfeba910d45562fe6b92ec3666aa94fb9155ee6c6ee9ab35f9e8988dce2898 instead of changing the interface. `teleport_robot` defined in SimulatorControl is a repeated field so we could put multiple commands there